Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.py

```diff
@@ -365,10 +341,48 @@ def get_file_request(self, sha256: List = None, expiration_seconds: int = None)
         return self._http_request('POST', suffix_url, json_data=body)
 
     def get_file_path_request(self, sha256: str = None) -> Dict:
-
         suffix_url = f'/ubs/v1/orgs/{self.cb_org_key}/sha256/{sha256}/summary/file_path'
         return self._http_request('GET', suffix_url)
 
+    def create_search_process_request(self, process_hash: str, process_name: str, event_id: str, query: str,
+                                      limit: int = None) -> dict:
+        if not process_hash and not process_name and not event_id and not query:
+            raise ValueError("No search parameters were provided.")
+        suffix_url = f'/api/investigate/v2/orgs/{self.cb_org_key}/processes/search_jobs'
+        process_hash_list = argToList(process_hash)
+        process_name_list = argToList(process_name)
+        body = assign_params(criteria=assign_params(
+            process_hash=process_hash_list,
+            process_name=process_name_list,
+            event_id=event_id,
+        ),
+            query=query,
+            rows=limit,
+            start=0
+
+        )
+        return self._http_request('POST', suffix_url, json_data=body)
+
+    def get_search_process_request(self, job_id) -> dict:
+        suffix_url = f'/api/investigate/v2/orgs/{self.cb_org_key}/processes/search_jobs/{job_id}/results'
+
+        return self._http_request('GET', suffix_url)
+
+    def create_search_event_by_process_request(self, process_guid: str, event_type: str,
+                                               query: str, limit: int = None) -> dict:
+        if event_type and event_type not in ['filemod', 'netconn', 'regmod', 'modload', 'crossproc', 'childproc']:
+            raise Exception('event_type is invalid')
```

Comment:
let's offer a solution, e.g:
```suggestion
            raise Exception("Only the following event types can be searched:  'filemod', 'netconn', 'regmod', 'modload', 'crossproc', 'childproc'")
```